### PR TITLE
Fix `series.str.startswith(tuple)`

### LINKED ISSUE #48575: BUG: 1.5rc raises TypeError when passing tuple to series.str.startswith()
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd

df = pd.util.testing.makeMixedDataFrame()

df.C.str.startswith(("a", "b"))
```

### Issue Description

Example works as expected in 1.4.4. After `pip install --upgrade --pre pandas==1.5.0rc0`, the same code raises

```py
TypeError: expected a string object, not tuple
```

<details>
<summary>stack trace</summary>

```py
      1 import pandas as pd
      3 df = pd.util.testing.makeMixedDataFrame()
----> 5 df.C.str.startswith(("a", "b"))

File ~/.venv/py310/lib/python3.10/site-packages/pandas/core/strings/accessor.py:129, in forbid_nonstring_types.<locals>._forbid_nonstring_types.<locals>.wrapper(self, *args, **kwargs)
    124     msg = (
    125         f"Cannot use .str.{func_name} with values of "
    126         f"inferred dtype '{self._inferred_dtype}'."
    127     )
    128     raise TypeError(msg)
--> 129 return func(self, *args, **kwargs)

File ~/.venv/py310/lib/python3.10/site-packages/pandas/core/strings/accessor.py:2345, in StringMethods.startswith(self, pat, na)
   2343 if not isinstance(pat, str):
   2344     msg = f"expected a string object, not {type(pat).__name__}"
-> 2345     raise TypeError(msg)
   2346 result = self._data.array._str_startswith(pat, na=na)
   2347 return self._wrap_result(result, returns_string=False)

TypeError: expected a string object, not tuple
```
</details>

I guess the ability to pass tuples was never documented? https://github.com/pandas-dev/pandas/issues/38281

### Expected Behavior

Should return boolean series.

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 224458ee25d92ccdf289d1ae2741d178df4f323e
python           : 3.10.6.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 21.6.0
Version          : Darwin Kernel Version 21.6.0: Wed Aug 10 14:28:23 PDT 2022; root:xnu-8020.141.5~2/RELEASE_ARM64_T6000
machine          : arm64
processor        : arm
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.UTF-8

pandas           : 1.5.0rc0
numpy            : 1.23.2
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 63.2.0
pip              : 22.1
Cython           : 0.29.26
pytest           : 7.1.2
hypothesis       : 6.42.0
...
xlrd             : 2.0.1
xlwt             : 1.3.0
zstandard        : None
tzdata           : None
</details>

### PR DESCRIPTION
**Pull request by janosh**
- [x] closes #48575 and #38281
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Should I add a note in `doc/source/whatsnew/v1.5.0.rst`?

### CODE REVIEW

File: pandas/core/strings/accessor.py [hunk 1]
```diff
@@ -2288,16 +2289,17 @@ def count(self, pat, flags=0):
         return self._wrap_result(result, returns_string=False)
 
     @forbid_nonstring_types(["bytes"])
-    def startswith(self, pat, na=None):
+    def startswith(self, pat: str | tuple[str, ...], na: Any = None) -> Series | Index:
```
Comment (mroeschke) [inline pandas/core/strings/accessor.py:2292]: Could `na` be typed `Scalar | None`?

**Review by mroeschke — COMMENTED** [review]
Thanks. Looks pretty good. No whatsnew needed.

Could you also add a tuple input example in the `Examples` section of the docstrings?

**Comment by janosh** [comment]
> Could you also add a tuple input example in the `Examples` section of the docstrings?

Certainly.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks for the quick fix @janosh!

**Comment by janosh** [comment]
@mroeschke You may want to close #38281 manually. I forgot to put a 'closes' in front of its PR number above.

**Pull request merged**